### PR TITLE
Completes task 6 retrieves comments by article id

### PR DESCRIPTION
**Pull request by davidhcodes**
Completes task 6 retrieves comments by article id, updates the endpoint.json to include the new endpoint

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -40,4 +40,37 @@ exports.fetchArticles = ()=>{
...
+
+    const queryValues = []
+
+       
+     let sqlQuery = `SELECT  comments.comment_id, articles.article_id, comments.votes, comments.created_at, comments.author, comments.body
+    FROM comments
+    INNER JOIN articles ON comments.article_id = articles.article_id `
+    
+
+    if(article_id){
```
Comment (nc-nick-jones) [inline models/articles.models.js:55]: don;t think this condition is necessary.  You are fetching by article ID and therefore should always have one.  So working with a complete SQL string should work fine.

File: __tests__/server.test.js [hunk 2]
```diff
@@ -144,4 +144,56 @@ describe("Testing APIs for GET /api/articles/:article_id", () => {
...
+          arrayOfComments.forEach((comment)=>{
+            expect(comment).toMatchObject({
+              author:expect.any(String),
+              comment_id:expect.any(Number),
+              body:expect.any(String),
+              article_id:expect.any(Number),
+              created_at:expect.any(String),
+              votes:expect.any(Number),
+            })
+          })
```
Comment (nc-nick-jones) [inline __tests__/server.test.js:172]: what about testing for 200 for an article with no comments?  this will require you to check if an article exists first. I think this was the topic of today's lecture.

**Review by nc-nick-jones — COMMENTED** [review]
Nice work David.  Just need to test for a 200 'returns empty array for article with no comments'.  To pass this you will need a way to check an article exists.  See comments for details.  good luck!
